lib/internal/http2/compat.js

```diff
@@ -428,20 +456,31 @@ class Http2ServerResponse extends Stream {
     stream.respond(headers);
   }
 
+  createPushResponse(headers, options, callback) {
+    var stream = this[kStream];
+    if (!stream) {
+      callback(new Error('HTTP/2 Stream has been closed'));
+      return;
+    }
+    return stream.pushStream(headers, options, callback);
+  }
+
   [kBeginSend]() {
     var state = this[kState];
     var stream = this[kStream];
     if (!state.headersSent) {
       state.headersSent = true;
-      const headers = llistToHeaders(this[kHeaders]);
+      const headers = this[kHeaders] || Object.create(null);
```

Comment:
TIL: The [nghttp2 `nghttp2_on_header_callback` API](https://nghttp2.org/documentation/types.html#c.nghttp2_on_header_callback) fires one header (name/value pair) at a time, so it is reasonable to append decoded headers to a (linked) list as they are received. This may explain why the linked list was used on the JS side.